pypit/artrace.py

```diff
@@ -294,42 +294,147 @@ def expand_slits(slf, mstrace, det, ordcen, extord):
     return lordloc, rordloc
 
 
-def trace_object(slf, det, sciframe, varframe, crmask, trim=2.0,
+def trace_objbg_image(slf, det, sciframe, slitn, objreg, bgreg, trim=2, triml=None, trimr=None):
+    """ Creates an image with weights corresponding to object or background pixels
+
+    Parameters
+    ----------
+    slf : Class instance
+      An instance of the Science Exposure class
+    det : int
+      Index of the detector
+    sciframe: numpy ndarray
+      Science frame
+    slitn : int
+      Slit (or order) number
+    objreg : list
+      A two element list containing arrays that indicate the left and right
+      pixels of the object locations for each object.
+    bgreg : list
+      A two element list containing arrays that indicate the background regions
+      around each object (i.e. [bgleft, bgright]). Each array is 2D and contains
+      a list of pixels for each object. The size of each array is [npix, nobj].
+    trim : int (optional)
+      Number of pixels to trim from the left and right slit edges.
+      To separately specify how many pixels to trim from the left
+      and right slit edges, use triml and trimr.
+    triml : int (optional)
+      Number of pixels to trim from the left slit edge
+    trimr : int (optional)
+      Number of pixels to trim from the right slit edge
+
+    Returns
+    -------
+    rec_obj_img : ndarray
+      An image containing weights to be used for the object
+    rec_bg_img : ndarray
+      An image containing weights to be used for the background
+    """
+    # Get the number of objects in the slit
+    nobj = len(objreg[0])
+    if triml is None:
+        triml = trim
+    if trimr is None:
+        trimr = trim
+    npix = int(slf._pixwid[det-1][slitn] - triml - trimr)
+    # Make an image of pixel weights for each object
+    xint = np.linspace(0.0, 1.0, sciframe.shape[0])
+    yint = np.linspace(0.0, 1.0, npix)
+    yint = np.append(-yint[1], np.append(yint, 2.0-yint[-2]))
+    msgs.info("Creating an image weighted by object pixels")
+    rec_obj_img = np.zeros((sciframe.shape[0], sciframe.shape[1], nobj))
+    for o in range(nobj):
```

Comment:
Ok, got it.

Close pairs of sources are a challenge (and one we will
deal with soon).